src/App.tsx

```diff
@@ -1,24 +1,128 @@
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState(Filter.ALL);
+  const [isHidden, setIsHidden] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+  const [todosError, setTodosError] = useState('');
+  const [processingTodoIds, setProcessingTodoIds] = useState<number[]>([]);
+  const [temporaryTodo, setTemporaryTodo] = useState<Todo | null>(null);
+
+  const loadTodos = async () => {
+    try {
+      const todosData = await getTodos(USER_ID);
+
+      setTodos(todosData);
+    } catch (error) {
+      setTodosError('Unable to load todos');
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
+  const visibleTodos = useMemo(() => {
+    switch (filterBy) {
+      case Filter.ACTIVE:
+        return todos.filter((todo) => !todo.completed);
+      case Filter.COMPLETED:
+        return todos.filter((todo) => todo.completed);
+      default:
+        return todos;
+    }
+  }, [todos, filterBy]);
+
+  const handleDelete = async (id: number) => {
+    setProcessingTodoIds(prev => [...prev, id]);
+
+    try {
+      const isTodoDelete = await deleteTodos(id);
+
+      if (isTodoDelete) {
+        setTodos(prev => prev.filter(todo => todo.id !== id));
+      } else {
+        setTodosError('Unable to delete a todo');
+      }
+    } catch (e) {
+      setTodosError('Unable to delete a todo');
+    } finally {
+      setProcessingTodoIds(prev => prev.filter(todoId => todoId !== id));
+    }
+  };
+
+  const handleTodoAdd = async (title: string) => {
+    setTemporaryTodo({
+      id: 0,
+      title,
+      completed: false,
+      userId: USER_ID,
+    });
+
+    try {
+      const createdTodo = await createTodo(title);
+
+      setTemporaryTodo(null);
+
+      setTodos(prevTodos => [...prevTodos, createdTodo]);
+    } catch (error) {
+      setTemporaryTodo(null);
+      setTodosError('Unable to add a todo');
+
+      throw new Error('Some error');
+    }
+  };
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <TodoappHeader
+          onTodoAdd={handleTodoAdd}
+          onError={setTodosError}
+        />
+
+        <TodoappList
+          todos={visibleTodos}
+          processingTodoIds={processingTodoIds}
+          handleDelete={handleDelete}
+          temporaryTodo={temporaryTodo}
+          setTodosError={setTodosError}
+        />
+
+        {/* Hide the footer if there are no todos */}
+        {visibleTodos.length > 0 && (
```

Comment:
If there is no `todos` at all, not `visibleTodos`. It works buggy. Try having only 1 todo, once you click on completed the footer will unmount